### DISCUSSION TITLE
Question:

### DISCUSSION THREAD

**Discussion by yaroslavkasatikov** [noobaa/noobaa-core#6631] [Q&A]
Hey team!

Could you please tell me if noobaa support Vhost Bucket style? I  mean https://bucketname.s3-noobaa.endpoint.domain.tld .
If not - is it in your future plans or not?

Thanks a lot,
Yaroslav

**Comment by jeniawhite** [comment]
@yaroslavkasatikov 
Hey,
We do support virtual host style buckets.

**Comment by yaroslavkasatikov** [comment]
Hi @jeniawhite !
Thanks for your reply.

is it enabled by default?
I tried to use pgBackRest in Crunchy PGO Operator which work only via vhost-style, but got 404 - bucket not found.

I did this steps:
1) I created bucket 'bucketname'
2) I created Openshift route "bucketname.s3-noobaa.apps.mucluster.mydomain.com" (default noobaa endpoint route is s3-noobaa.apps.mucluster.mydomain.com) and issue correct TLS certificate

Is it correct actions or I need to do something additional?

**Comment by guymguym** [comment]
Hi @yaroslavkasatikov 
Can you take a look on this doc and let us know if that helps? Otherwise we might need to improve it - https://github.com/noobaa/noobaa-operator/blob/master/doc/ssl-dns-routing.md
Thanks

**Comment by guymguym** [comment]
Oh I see you create a route per your bucket.
Adding @nb-ohad - do we need to create a route per bucket to use bucket endpoints?

**Comment by yaroslavkasatikov** [comment]
Team,
do you have any updates here?

**Comment by nb-ohad** [comment]
I see no need for a route for each bucket.

We support by default: `<bucket>.<s3 route DNS name>` for clients outside the cluster and  `<bucket>.<internal services DNS name>` for clients that run from a pod inside the cluster.

Now you can always add other suffixes to the virtual host list in the noobaa CR (in the spec section), but it is your responsibility to set up name resolution in a way that `*.<the suffix you choose>` resolve to a reachable address (from your client) to the noobaa s3 service.

I hope this clarifies the subject.
